DataFormats/simulation/src/Stack.cxx

```diff
@@ -312,32 +313,13 @@ void Stack::UpdateTrackIndex(TRefArray *detList)
   }
 
   FairDetector *det = nullptr;
-  while ((det = (FairDetector *) fDetIter->Next())) {
-
-    // Get hit collections from detector
-    Int_t iColl = 0;
-    TClonesArray *hitArray;
-    while ((hitArray = det->GetCollection(iColl++))) {
-      nColl++;
-      Int_t nPoints = hitArray->GetEntriesFast();
-
-      // Update track index for all MCPoints in the collection
-      for (Int_t iPoint = 0; iPoint < nPoints; iPoint++) {
-        auto *point = (o2::BaseHit *) hitArray->UncheckedAt(iPoint);
-        Int_t iTrack = point->GetTrackID();
-
-        mIndexIterator = mIndexMap.find(iTrack);
-        if (mIndexIterator == mIndexMap.end()) {
-          if (mLogger) {
-            mLogger->Fatal(MESSAGE_ORIGIN, "Stack: Track index %i not found in index map! ", iTrack);
-          }
-          Fatal("Stack::UpdateTrackIndex", "Track index not found in map");
-        }
-        point->SetTrackID((*mIndexIterator).second);
-      }
+  while ((det = (FairDetector*)fDetIter->Next())) {
```

Comment:
is the cast to `FairDetector` still needed, or would
`while ( (auto det = fDetIter->Next()) )` suffice?

Reply:
you are right. Will fix this as well.